### ISSUE TITLE
Boolean Indexing not supported on DataFrame

### ISSUE THREAD

**Issue by muellren** [rapidsai/cudf#515]
## Reporting a bug

Boolean Indexing does not work on cudf.DataFrame() as described in 
https://rapidsai.github.io/projects/cudf/en/latest/10min.html#selection
Reporting as a bug in the documentation. 

```python
gdf = cudf.DataFrame()
gdf['col1'] = [1, 3, 2, 3]
gdf['col2'] = [4.0, 2.0, 1.0, 0.0]
filtered_gdf = gdf[gdf.col1 == 3]
```
Results:
```
TypeError                                 Traceback (most recent call last)
<ipython-input-139-d5cead4f8e8a> in <module>
      2 gdf['col1'] = [1, 3, 2, 3]
      3 gdf['col2'] = [4.0, 2.0, 1.0, 0.0]
----> 4 filtered_gdf = gdf[gdf.col1 == 3]

~/anaconda3/envs/cudf_dev/lib/python3.5/site-packages/cudf-0.4.0-py3.5-linux-x86_64.egg/cudf/dataframe/dataframe.py in __getitem__(self, arg)
    231         else:
    232             msg = "__getitem__ on type {!r} is not supported"
--> 233             raise TypeError(msg.format(arg))
    234 
    235     def __setitem__(self, name, col):

TypeError: __getitem__ on type <cudf.Series nrows=4 > is not supported
```

- [X] I am using the latest version of cuDF from conda, built from `master`, or
built from the latest tagged release: master
- [X] I have included the version or commit hash I am using for reference (commit 7aa3799).
- [X] I have included the following environment details: Ubuntu 16.04, 4.15.0-4-generic, GTX 1080
- [X] I have included the following version information for:
      Arrow (0.10.0), cmake (3.13.1), CUDA (9.2.148), gcc/g++ (5.4.0), Numpy, Pandas (0.23.4), Python (3.5.5)
- [X] I have included below a minimal working reproducer

**Comment by kkraus14** [comment]
Note: the docs show boolean indexing on a series, not on the dataframe and I believe we support such currently. We should support boolean indexing on the dataframe level as well.

**Comment by muellren** [comment]
I see, indeed. The documentation only shows the bitvector applied to a series. 

I can confirm that 
```python
gdf = cudf.DataFrame()
gdf['col1'] = [1, 3, 2, 3]
gdf['col2'] = [4.0, 2.0, 1.0, 0.0]
filtered_col2 = gdf.col2[gdf.col1 == 3]
``` 
does work.

This is not a bug, rather a feature request then.

**Comment by beckernick** [comment]
Agreed. The query API works well but adds complexity compared to direct indexing. Would be a nice feature for those used to pandas.

**Comment by kkraus14** [comment]
@thomcom would you want to try to tackle this? I believe it should be as simple as applying the bitmask to each column and assigning them to a new dataframe object.

**Comment by kkraus14** [comment]
Fixed by #524
